### ISSUE TITLE
Pull Request button doesn't reflect selected branch for fork

### ISSUE THREAD

**Issue by rbukovansky** [go-gitea/gitea#304]
- Gitea version (or commit ref): 42ec5ce740fd
- Git version: 1.8.3.1
- Operating system: CentOS 7.2
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] SQLite
- Log gist:

## Description
1. Make Fork->Clone->Branch->Push cycle, so you end up with new branch pushed to your remote repository
2. Open your Gitea and go to your fork
3. Select pushed branch
4. Click on Pull Request button.

Gitea doesn't reflect selected branch in link of the PR button, so it always tries to create PR for i.e. `<upstream>:master...<fork>:master`, instead of `<upstream>:master...<fork>:<pushed_and_selected_branch>`. So I need then to change the branch for my fork after I click the PR button, which I find annoying as I have already selected the branch I want to have that PR from.

Please, correct it. Thanks.

Same issue as https://github.com/gogits/gogs/issues/3604

**Comment by bkcsoft** [comment]
this should be a simple matter of changing from `.Repo.DefaultBranch` to `ctx.Repo.CurrentBranch` (or something similar)

**Comment by lunny** [comment]
@bkcsoft, send a PR?

**Comment by rbukovansky** [comment]
I've got some workaround for this changing one line in `templates/repo/home.tmpl` from this:
```html
<a href="{{.BaseRepo.Link}}/compare/{{.BaseRepo.DefaultBranch}}...{{.PullRequestCtx.HeadInfo}}">
```
to this:
```html
<a href="{{.BaseRepo.Link}}/compare/{{.BaseRepo.DefaultBranch}}...{{.Owner.Name}}:{{.BranchName}}">
```
but investigating further it seems to me that this line:
```go
ctx.Repo.PullRequest.HeadInfo = ctx.Repo.Owner.Name + ":" + ctx.Repo.BranchName
```
in modules/context/repo.go is not working properly.

But I'm not Go guy, so I can't debug it by myself.

**Comment by lunny** [comment]
closed by #762
